osf/models/notifications.py

```diff
@@ -75,7 +75,8 @@ def add_user_to_subscription(self, user, notification_type, save=True):
                 self.owner.parent_node.save()
 
         if save:
-            self.save()
+            # Do not clean legacy objects
+            self.save(clean=False)
```

Comment:
Consider putting in a try block where we run the clean by default and then check for the specific issue and save without `clean`ing only if we know it's an existing exception

Reply:
This can be a follow-up ticket